airflow-core/src/airflow/ui/src/layouts/Details/Graph/Graph.tsx

```diff
@@ -94,11 +94,14 @@ export const Graph = () => {
     [graphData.nodes],
   );
 
+  const prevObservedGroupIds = useRef<Array<string>>([]);
+
   useEffect(() => {
-    if (observedGroupIds !== allGroupIds) {
+    if (JSON.stringify(observedGroupIds) !== JSON.stringify(prevObservedGroupIds.current)) {
```

Comment:
Even if we do that sometimes for comparing complex object (I think we really shouldn't because things will fail for circular references), it's really easy to do a proper comparison here for simple array of int object. At some point we might want to install lodash library to defer all that logic to a helper package.